solutions/Kattis Robot Turtles.mdx

```diff
@@ -0,0 +1,135 @@
+---
+id: kattis-robotturtles
+source: Kattis
+title: Robot Turtles
+author: Maggie Liu
+---
+
+## Explanation
+
+We can use Dijkstra's, keeping track of the current length of the program
+($\texttt{dist}$), the direction the turtle faces ($\texttt{dir}$) and the moves
+it has already made at each location ($\texttt{moves}$). If there is an ice
+castle in front of the turtle, it should only be melted if the turtle will move
+there immediately. If the turtle doesn't move onto the square immediately, then
+it will either:
+
+- return to the square later, in which case the turtle has wasted moves before
+  returning.
+- never move onto the square, in which case the turtle wasted a move melting the
+  ice castle. 
+So, we can only melt ice castles if the turtle moves forward immediately after.
+
+There are 3 moves that a turtle can make at each location. It can move forward
+if there is an empty square, or melt an ice castle and then move forward if
+there is an ice castle. Or, the turtle can turn left or turn right. As we make
+moves, we add the instructions to the string $\texttt{moves}$.
+
+Once we reach the diamond, we can print out the $\texttt{moves}$.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N^2\log N^2)$, where $N$ is the side length of the grid.
```

Comment:
is this right? or would it be N^2 \log N since the 2 is a constant factor?

Reply:
i think this is right